targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/TARGET_FRDM/PeripheralPins.c

```diff
@@ -17,7 +17,16 @@
 #include "PeripheralPins.h"
 #include "PeripheralPinMaps.h"
 
-const PinMap *gpio_pinmap()
+// List of GPIOs with limited functionality
+const PinList *pinmap_gpio_restricted_pins()
 {
-    return PinMap_GPIO;
+    static const PinName pins[] = {
+        PTE24,  // fixed pull-up (for I2C)
+        PTE25,  // fixed pull-up (for I2C)
+    };
+    static const PinList pin_list = {
+        sizeof(pins) / sizeof(pins[0]),
+        pins
+    };
+    return &pin_list;
```

Comment:
Note, there is an overlap with PR #12477 (https://github.com/ARMmbed/mbed-os/pull/12477/commits/1b894aa67ad468eb16a326da34f1402552c5d9b3#diff-22a0fab58312a8cd5c56864402f4aa94). If #12477 gets merged first, we do not need this implementation of `pinmap_gpio_restricted_pins()` for K64F any more. The input pull mode tests will be skipped based on pin's capabilities now.

Reply:
Thanks for the clarification. In this case, I think #12477 should go first. And then I rebase and update this one.